### PR TITLE
ENH: Add annotations for `np.lib.ufunclike`

### PR DESCRIPTION
**Pull request by BvB93**
Per the title: this PR adds annotations for `np.lib.ufunclike`.

DType support also turned out to be fairly straightforward to implement here, so that was definitely a plus.

### CODE REVIEW

File: numpy/lib/ufunclike.py [hunk 1]
```diff
@@ -189,7 +189,8 @@ def isposinf(x, out=None):
     try:
         signbit = ~nx.signbit(x)
     except TypeError as e:
-        raise TypeError('This operation is not supported for complex values '
+        dtype = nx.asanyarray(x).dtype
+        raise TypeError(f'This operation is not supported for {dtype} values '
```
Comment (BvB93) [inline numpy/lib/ufunclike.py:193]: The exception message would previously assume that that the dtype of the (invalid) array was complex. This is not necessarily true though, as the very same exception will also be triggered by `timedelta64` and `datetime64` arrays.

File: numpy/lib/ufunclike.pyi [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+_ArrayType = TypeVar("_ArrayType", bound=ndarray[Any, Any])
+
+__all__: List[str]
+
+@overload
+def fix(
+    x: _ArrayLikeFloat_co,
+    out: None = ...,
+) -> _ArrayOrScalar[floating[Any]]: ...
+@overload
```
Comment (charris) [inline numpy/lib/ufunclike.pyi:19]: Be good to have 2 blank lines between functions (PEP8).

Comment (charris) [reply numpy/lib/ufunclike.pyi:19]: I know you are trying to group the functions, but I find it more difficult to read without the blank lines.

**Comment by charris** [comment]
Just some style nits, otherwise LGTM.

File: numpy/lib/ufunclike.pyi (hunk 2 above)
Comment (BvB93) [reply numpy/lib/ufunclike.pyi:19]: For overloads it's generally common practice to exclude the blank lines ([example from typeshed](https://github.com/python/typeshed/blob/8f7e2216b66b9e8181b119fa2f7d560fca987605/stdlib/builtins.pyi#L677-L680)).

Note that this area is not quite covered by PEP 8 though: the description of a single function with multiply signatures.

**Comment by charris** [comment]
Thanks Bas.

**Pull request merged**